Sources/DatadogCrashReporting/PLCrashReporterIntegration/DDCrashReportExporter.swift

```diff
@@ -0,0 +1,190 @@
+                .compactMap { $0.value as? UnsafePointer<Int8> }
+                .map { String(cString: $0).uppercased() } // [HUP, INT, QUIT, ILL, TRAP, ABRT, ...]
+
+            let knownSignalDescriptions = Mirror(reflecting: sys_siglist)
+                .children
+                .compactMap { $0.value as? UnsafePointer<Int8> }
+                .map { String(cString: $0) } // [Hangup, Interrupt, Quit, Illegal instruction, ...]
+
+            if knownSignalNames.count == knownSignalDescriptions.count { // sanity check
+                if let index = knownSignalNames.firstIndex(where: { signalName == "SIG\($0)" }) {
+                    let signalDescription = knownSignalDescriptions[index]
+                    return "Application crash: \(signalName) (\(signalDescription))"
+                }
+            }
+
+            return "Application crash: \(unknown)"
+        }
+    }
+
+    /// Formats the error stack - in Datadog Error Tracking this corresponds to `error.stack`.
+    ///
+    /// **Note:** This produces unsymbolicated stack trace, which is later symbolicated backend-side and used for building error's fingerprint in Error Tracking.
+    private func formatStack(for crashReport: CrashReport) -> String {
+        let crashedThread = crashReport.threads.first { $0.crashed }
+        let exception = crashReport.exceptionInfo
+
+        // Consider most meaningful stack trace in this order:
+        // - uncaught exception stack trace (if available)
+        // - crashed thread stack trace (must be available)
+        // - first thread stack trace (sanity fallback)
+        let mostMeaningfulStackFrames = exception?.stackFrames
+            ?? crashedThread?.stackFrames
+            ?? crashReport.threads.first?.stackFrames
+
+        guard let stackFrames = mostMeaningfulStackFrames else {
+            return unavailable // should never be reached
+        }
+
+        return string(from: stackFrames)
+    }
+
+    // MARK: - Exporting threads and binary images
+
+    private func threads(from crashReport: CrashReport) -> [DDCrashReport.Thread] {
+        return crashReport.threads.map { thread in
+            return DDCrashReport.Thread(
+                name: "Thread \(thread.threadNumber)",
+                stack: string(from: thread.stackFrames),
+                crashed: thread.crashed,
+                state: nil // TODO: RUMM-1462 Send registers state for crashed thread
+            )
+        }
+    }
+
+    private func binaryImages(from crashReport: CrashReport) -> [DDCrashReport.BinaryImage] {
+        return crashReport.binaryImages.map { image in
+            // Ref. for this computation:
+            // https://github.com/microsoft/plcrashreporter/blob/dbb05c0bc883bde1cfcad83e7add25862c95d11f/Source/PLCrashReportTextFormatter.m#L447
+            let loadAddressHex = "0x\(image.imageBaseAddress.toHex)"
+            var maxAddressOffset = image.imageSize.subtractIfNoOverflow(1) ?? image.imageSize
```

Comment:
`imageSize` is an unsigned integer and here we check for reaching overflow below `0` (by subtracting `1` from `UInt`). Although images cant have negative size and (perhaps) won't have the size of `0`, this is a sanity check to avoid overflow crash if sth goes wrong. We retrieve this `image.imageSize` value from `PLCrashReporter` (3rd party), which reads it from the system (it's read from `C` to `Obj-C` to `Swift`), so IMO extra safety is a good idea here 🙂.

Reply:
Tks for the explanation, yes I do agree that extra safety will not harm.